packages/rust-bindings/src/aux_functions.rs

```diff
@@ -821,6 +822,76 @@ pub fn create_id_proof_aux(input: IdProofInput) -> Result<JsonString> {
+#[serde(rename_all = "camelCase", tag = "type")]
+pub enum DeserializableCommitmentInputs {
+    #[serde(rename_all = "camelCase")]
+    Account {
+        /// Issuance date of the credential that the proof is about.
+        /// This is an unfortunate name to conform to the standard, but the
+        /// meaning here really is `validFrom` for the credential.
+        issuance_date: String,
+        issuer:        IpIdentity,
+        /// The values that are committed to and are required in the proofs.
+        #[serde_as(as = "BTreeMap<serde_with::DisplayFromStr, _>")]
+        values:        BTreeMap<u8, Web3IdAttribute>,
+        /// The randomness to go along with commitments in `values`.
+        #[serde_as(as = "BTreeMap<serde_with::DisplayFromStr, _>")]
+        randomness:    BTreeMap<u8, PedersenRandomness<constants::ArCurve>>,
+    },
+    /// Inputs are for a credential issued by Web3ID issuer.
+    #[serde(rename_all = "camelCase")]
+    Web3Issuer {
+        /// Issuance date of the credential that the proof is about.
+        /// This is an unfortunate name to conform to the standard, but the
+        /// meaning here really is `validFrom` for the credential.
+        issuance_date: String,
+        #[serde(deserialize_with = "base16_decode")]
+        /// The signer that will sign the presentation.
+        signer: ed25519_dalek::Keypair,
+        /// Values that are committed to.
+        #[serde_as(as = "BTreeMap<serde_with::DisplayFromStr, _>")]
+        values:        BTreeMap<u8, Web3IdAttribute>,
+        /// The randomness for the vector commitment.
+        randomness:    PedersenRandomness<constants::ArCurve>,
+    },
+}
+
+pub fn create_web3_id_proof_aux(input: Web3IdProofInput) -> Result<JsonString> {
+    let mut commitment_inputs: Vec<CommitmentInputs<'_, constants::ArCurve, Web3IdAttribute, ed25519_dalek::Keypair>> = Vec::new();
+    for ref inp in input.commitment_inputs.iter() {
+        commitment_inputs.push(match inp {
+            DeserializableCommitmentInputs::Account{ issuance_date, issuer, values, randomness} => {
+                CommitmentInputs::Account{
+                    issuance_date: chrono::DateTime::parse_from_rfc3339(&issuance_date).unwrap().with_timezone(&chrono::Utc),
+                    issuer: *issuer,
+                    values,
+                    randomness,
+                }
+            },
+            DeserializableCommitmentInputs::Web3Issuer{ issuance_date, signer, values, randomness} => {
+                CommitmentInputs::Web3Issuer{
+                    issuance_date: chrono::DateTime::parse_from_rfc3339(&issuance_date).unwrap().with_timezone(&chrono::Utc),
+                    signer,
+                    values,
+                    randomness: randomness.clone(),
+                }
+            }
+        });
+    }
+    let presentation = input.request.prove(&input.global_context, commitment_inputs.into_iter());
+    Ok(json!(presentation.unwrap()).to_string())
+}
```

Comment:
@abizjak This is how I "had" to do it. :thinking:

Reply:
Shouldn't we just add serde (de)serialization to commitmentinputs directly?

Why is it necessary to have this extra type?

Reply:
I guess we can, we just need to change CommitmentInputs to not contain references to the fields. I can probably make a PR for that.

Reply:
Right, but I like the references there. Especially for the secret key so as to avoid copying it.

That was the rationale for using them.